pkg/api/types.go

```diff
@@ -200,6 +200,8 @@ type VolumeSource struct {
 	ISCSI *ISCSIVolumeSource `json:"iscsi"`
 	// Glusterfs represents a Glusterfs mount on the host that shares a pod's lifetime
 	Glusterfs *GlusterfsVolumeSource `json:"glusterfs"`
+	// CinderPersistentDisk represents a cinder volume attached and mounted on kubelets host machine
+	CinderPersistentDisk *CinderPersistentDiskVolumeSource `json:"cinderPersistentDisk"`
```

Comment:
I'm not sure we want this as a VolumeSource long term - PersistentVolumeSource is a better abstraction.  In the short term, I'll review the rest of this PR and come back to this point :)